pgmpy/tests/test_models/test_MarkovModel.py

```diff
@@ -115,6 +115,25 @@ def test_get_cardinality(self):
         self.graph.remove_factors(phi1, phi2, phi3)
         self.assertDictEqual(self.graph.get_cardinality(), {})
 
+    def test_get_cardinality_with_node(self):
+
+        self.graph.add_edges_from([('a', 'b'), ('b', 'c'), ('c', 'd'),
+                                   ('d', 'a')])
+
+        phi1 = DiscreteFactor(['a', 'b'], [1, 2], np.random.rand(2))
+        self.graph.add_factors(phi1)
+        self.assertEqual(self.graph.get_cardinality('a'), 1)
+        self.assertEqual(self.graph.get_cardinality('b'), 2)
+        self.graph.remove_factors(phi1)
+
+        phi1 = DiscreteFactor(['a', 'b'], [2, 2], np.random.rand(4))
```

Comment:
Both these use cases seem the same to me. Let me know if I am wrong otherwise just keep one of these.